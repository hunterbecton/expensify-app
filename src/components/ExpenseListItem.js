import React from 'react';
import { Link } from 'react-router-dom';


const ExpenseListItem = ({description, amount, createdAt, id, dispatch}) => (
    <div>
        <Link to={`/edit/${id}`}><h3>{ description }</h3></Link>
        <p>{ amount } - { createdAt }</p>
    </div>
);

//Nothing is needed form the state, which is why we don't pass anything into the connect function
export default ExpenseListItem;