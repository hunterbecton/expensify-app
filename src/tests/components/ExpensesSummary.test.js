import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should correctly render ExpensesSummary with one expense',() => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={235} />)
    expect(wrapper).toMatchSnapshot();
});

test('Should correctly render ExpensesSummary with multiple expense',() => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expenseTotal={235977943743} />)
    expect(wrapper).toMatchSnapshot();
});