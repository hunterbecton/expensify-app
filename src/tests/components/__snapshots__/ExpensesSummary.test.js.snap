// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render ExpensesSummary with multiple expense 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $2,359,779,437.43
     
  </h1>
</div>
`;

exports[`Should correctly render ExpensesSummary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
     
  </h1>
</div>
`;
